virttest/qemu_devices/qcontainer.py

```diff
@@ -2935,22 +2934,45 @@ def dimm_device_define_by_params(self, params, name):
 
     def memory_define_by_params(self, params, name):
         """
-        Create memory modules by params, include memory object and
-        pc-dimm devices.
+        Create memory modules by params, include memory object,
+        pc-dimm, nvdimm and virtio-mem devices.
         """
         params = params.object_params(name)
         devices = []
         if not self.has_device("pc-dimm"):
             LOG.warn("'PC-DIMM' does not support by your qemu")
             return devices
+        mem_devtype = params.get("mem_devtype", "dimm")
```

Comment:
Suggest `vm_memdev_model` as the name for this new param, because:

1. to avoid the potential naming collisions it'd better to have `vm` as the namespace
2. to align to the naming style of nic devices (`nic_model`)

Reply:
Included now, thanks!